namer/namer.cc

```diff
@@ -1432,27 +1434,28 @@ class SymbolDefiner {
             sym = existingTypeMember;
             sym.data(ctx)->addLoc(ctx, ctx.locAt(typeMember.asgnLoc));
         } else {
-            auto name = typeMember.name;
-            auto oldSym = onSymbol.data(ctx)->findMemberNoDealias(ctx, name);
+            auto oldSym = onSymbol.data(ctx)->findMemberNoDealias(ctx, typeMemberName);
             if (oldSym.exists()) {
                 emitRedefinedConstantError(ctx, typeMember.nameLoc, oldSym.name(ctx), core::SymbolRef::Kind::TypeMember,
                                            oldSym);
-                name = ctx.state.nextMangledName(onSymbol, name);
+                typeMemberName = ctx.state.nextMangledName(onSymbol, typeMemberName);
             }
-            sym = ctx.state.enterTypeMember(ctx.locAt(typeMember.asgnLoc), onSymbol, name, variance);
+            sym = ctx.state.enterTypeMember(ctx.locAt(typeMember.asgnLoc), onSymbol, typeMemberName, variance);
 
             // The todo bounds will be fixed by the resolver in ResolveTypeParamsWalk.
             auto todo = core::make_type<core::ClassType>(core::Symbols::todo());
             sym.data(ctx)->resultType = core::make_type<core::LambdaParam>(sym, todo, todo);
+        }
 
-            if (isTypeTemplate) {
-                auto typeTemplateAliasName = typeMember.name;
-                auto context = ctx.owner.enclosingClass(ctx);
+        if (isTypeTemplate) {
+            auto typeTemplateAliasName = typeMember.name;
+            auto context = ctx.owner.enclosingClass(ctx);
+            if (!existingTypeMember.exists()) {
```

Comment:
What do you think about flipping this condition, so that it's the same as the `existingTypeMember.exists()` condition above?

Reply:
No problem. 
I originally decided to make it `!existingTypeMember.exists()`, because it fits the journey of type template better. At the first run of namer type member doesn't exist, so we create a new static field (first branch). On the second run and after it for the same type member we would use only the second branch. But yeah, I'm not holding to that